### ISSUE TITLE
Core: Switch to Object.defineProperty(` this, 'is*', { value: true } );

### ISSUE THREAD

**Issue by RenaudRohlinger** [mrdoob/three.js#24093]
Tried give a try to the `this.is` commit, everything looks except when serializing / deserializing ```TypeError: Cannot add property isVector3, object is not extensible```

I had to convert all the math class to this pattern: 
```Object.defineProperty(` this, 'isVector3', { value: true } );```

Related:
#24047 #24092

**Comment by Mugen87** [comment]
> except when serializing / deserializing

Um, I don't see any problems with the editor. How can the error be reproduced?

**Comment by Mugen87** [comment]
> I had to convert all the math class to this pattern: Object.defineProperty(` this, 'isVector3', { value: true } );

Probably no good idea: #21284.

**Comment by RenaudRohlinger** [comment]
Ah I see, indeed. that was when sending `vec3` via postMessage with service workers. I will try to investigate on that end then.

**Comment by marcofugaro** [comment]
> Probably no good idea: https://github.com/mrdoob/three.js/issues/21284.

Yup, we used `Object.defineProperty()` in the past but turns out it was slow.

**Comment by Mugen87** [comment]
@marcofugaro I've just noticed that _public instance fields_ are internally added via `Object.defineProperty()`, see [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Public_class_fields#public_instance_fields). 

> Public instance fields are added with Object.defineProperty() either at construction time in the base class (before the constructor body runs), or just after super() returns in a subclass.

That means _public instance fields_ could eventually introduce the same performance issue like when directly using `Object.defineProperty()`. If we convert `.is*` flags to _public instance fields_ in the future, we should definitely have an eye on the performance.

**Comment by marcofugaro** [comment]
@Mugen87 ah yes good point, we need to do performance profiling before switching to class fields. And report eventual finding as you said.
